frontend/helpers/path-helpers.ts

```diff
@@ -19,12 +19,17 @@ type ProductListPathAttributes = Pick<
    ProductList,
    'type' | 'handle' | 'deviceTitle'
 >;
-
-export function productListPath(
-   productList: ProductListPathAttributes,
-   itemType?: string,
-   variant?: string | undefined
-): string {
+type ProductListPathProps = {
+   productList: ProductListPathAttributes;
+   itemType?: string;
+   variant?: string | undefined;
+};
```

Comment:
Adding to my previous comment, it seems to me that this helper is becoming increasingly complex, and some of that complexity could be simplified if pushed into the model.
Ideally one should be able to call `productListPath(productList)` and get back the product list path. 
`itemType` is just a specific type of product list, not an additional attribute I need to think about when I'm trying to get the product list path.
Maybe we can conceive a similar simplification for `variant`? When should I pass this param to `productListPath`? Can we make it so when using `productListPath` we don't have to think about this?